dispatcher_plugin_nb2workflow/products.py

```diff
@@ -174,6 +174,27 @@ def write(self):
     def get_html_draw(self):
         return {'image': {'div': '<br>'+self.data_prod, 'script': ''} }
 
+
+class NB2WProgressProduct(NB2WProduct):
+    # TODO to be adapted
+    type_key = 'http://odahub.io/ontology#ODAProgressProduct'
+
+    def __init__(self, progress_html_data, out_dir=None, name='progress'):
+        self.out_dir = out_dir
+        self.name = name
+        self.progress_data = progress_html_data
+
+    def write(self):
+        file_path = os.path.join(self.out_dir, self.name)
+        with open(file_path, 'w') as fd:
+            fd.write(self.progress_data)
+        self.file_path = file_path
+
+    def get_html_draw(self):
+        # TODO not sure is correct, to verify
+        return {'image': {'div': '<br>' + self.progress_data, 'script': ''}}
```

Comment:
This is not the standard product, so it will be treated by frontend in a special way, not embedded in the product view modal window. Probably, opening the html representation of the notebook in the new browser tab will be enough. If so, we can get rid of these methods here (will require an adaptation of `NB2WProductQuery.process_product_method()`)

Reply:
which type of adaption? in `process_product_method`, in case of a progressProduct product, those two methods are not used

Reply:
It's treated here https://github.com/oda-hub/dispatcher-plugin-nb2workflow/blob/908cda39605dc3f62f3fc9773374b1c13cda47de/dispatcher_plugin_nb2workflow/queries.py#L164-L167 in case of oda_api request.

If it's frontend request, the processing is in this `else` clause https://github.com/oda-hub/dispatcher-plugin-nb2workflow/blob/908cda39605dc3f62f3fc9773374b1c13cda47de/dispatcher_plugin_nb2workflow/queries.py#L176
and there is no special treatment added there for this product.

Reply:
But we can, of course, set `api=true` for this specific request even in frontend and not bother about this stuff

Reply:
I think it's better if I adapt the function accordingly, it will be more controllable from the side of the plugin

Reply:
`process_product_method` is now adapted, and `write` and `process_product_method` can actually be removed...

What do you think?

Reply:
Yeah, let's remove them